scenarios/GitHubLovesSparkDotNet/GitHubLovesSparkDotNet/Program.cs

```diff
@@ -0,0 +1,124 @@
+    // Acknowledgements:
+    // - The GHTorrent project
+    //   http://ghtorrent.org
+    // - Azure Databricks 
+    //   https://azure.microsoft.com/en-us/services/databricks/
+    // - Azure Data Lake Storage Gen2 (ABFS) 
+    //   https://azure.microsoft.com/en-us/services/storage/data-lake-storage/
+
+    class Program
+    {
+        public static string StorageConfigKey => "fs.azure.account.key.<your-storage-account-name>.dfs.core.windows.net";
+        public static string SecureStorageKey => "<your-storage-account-key>";
+        public static string DataStoragePath => "abfss://<filesystem>@<storage-account>.dfs.core.windows.net/<path>/";
+
+        static void Main(string[] args)
+        {
+            var spark =
+                SparkSession
+                    .Builder()
+                    .Config(StorageConfigKey, SecureStorageKey)
+                    .AppName(@"Github 💖 .NET for Apache Spark")
+                    .GetOrCreate();
+
+            // Initialize all dataframes (which point to CSV files on the storage system)
+            DataFrame commits = 
+               spark.Read()
+                    .Schema("id INT, sha STRING, author_id INT, committer_id INT, " +
+                            "project_id INT, created_at TIMESTAMP")
+                    .Csv(DataStoragePath + "commits.csv");
+
+            DataFrame watchers = 
+               spark.Read()
+                    .Schema("repo_id INT, user_id INT, created_at TIMESTAMP")
+                    .Csv(DataStoragePath + "watchers.csv");
+
+            DataFrame projects = 
+               spark.Read()
+                    .Schema("id INT, url STRING, owner_id INT, name STRING, " + 
+                            "descriptor STRING, language STRING, created_at STRING, " + 
+                            "forked_from INT, deleted STRING, updated_at STRING")
+                    .Csv(DataStoragePath + "projects.csv")
+                    .Filter(Col("language") == "C#");
+
+            // Use functional programming to find top C# projects by stars
+            DataFrame stars =
+              projects
+              .Join(watchers, Col("id") == watchers["repo_id"])
+              .GroupBy("name")
+              .Agg(Count("*").Alias("stars"))
+              .OrderBy(Desc("stars"));
+
+            stars.Show();
+
+            // Or... Spark SQL!!!
+            projects.CreateOrReplaceTempView("projects");
+            watchers.CreateOrReplaceTempView("watchers");
+            spark.Sql(@"
+                SELECT name, COUNT(*) AS stars 
+                | FROM projects 
+                | INNER JOIN watchers w 
+                | ON id = w.repo_id 
+                | GROUP BY name 
+                | ORDER BY stars DESC".StripMargin()).Show();
+
+            // Let's figure out the top projects (w.r.t., stars)
+            // and how developer commit pattern looks like over a week - 
+            // do people work more over weekdays or weekends? :)
+            var p =
```

Comment:
what is p? also `DataFrame` to be consistent.

Reply:
Done